Shopify/ruby-lsp
# Always push telemetry/event notifications

### Motivation

This work is necessary to help address the problem described in https://github.com/Shopify/vscode-ruby-lsp/pull/85.

It's brittle to let the plugin tell the server if `telemetry/event` notifications should be pushed or not. It's more robust to always send the notifications and just let the plugin decide whether to discard or do something with them.

### Implementation

Removed the check for whether telemetry was enabled or not and also removed the ignored requests logic. It's much simpler to just send everything to the plugin and let it decide.

### Automated Tests

Added the extra assertions expected from always sending telemetry notifications.

### Manual Tests

1. Start the plugin in debug mode using https://github.com/Shopify/vscode-ruby-lsp/pull/85 and this PR's branch for the server
2. Verify that telemetry events are printed in the debug console of the plugin window
